src/python/gudhi/representations/metrics.py

```diff
@@ -116,33 +192,54 @@ def transform(self, X):
         Returns:
             numpy array of shape (number of diagrams in **diagrams**) x (number of diagrams in X): matrix of pairwise bottleneck distances.
         """
-        num_diag1 = len(X)
+        if not USE_GUDHI:
+            print("Gudhi built without CGAL: returning a null matrix")
+        Xfit = pairwise_persistence_diagram_distances(X, self.diagrams_, metric="bottleneck", e=self.epsilon) if USE_GUDHI else np.zeros((len(X), len(self.diagrams_)))
+        return Xfit
 
-        #if len(self.diagrams_) == len(X) and np.all([np.array_equal(self.diagrams_[i], X[i]) for i in range(len(X))]):
-        if X is self.diagrams_:
-            matrix = np.zeros((num_diag1, num_diag1))
+class WassersteinDistance(BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing the Wasserstein distance matrix from a list of persistence diagrams. 
+    """
+    def __init__(self, order=2, internal_p=2, mode="pot", delta=0.0001):
```

Comment:
I am not sure about having different default values of delta everywhere, unless there is some particular reason why a different default value makes sense there. If you think the default value in hera.cc is not good, we can change it, no problem.

Reply:
I agree. I modified to same default value as in `hera.cc`.